systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java

```diff
@@ -100,20 +109,120 @@ void testRackAware() {
+                .withListenerName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+                .build();
+
+        internalKafkaClient.verifyProducedAndConsumedMessages(
+                internalKafkaClient.sendMessagesPlain(),
+                internalKafkaClient.receiveMessagesPlain()
         );
     }
 
+    @Test
+    @Tag(CONNECT)
+    @Tag(REGRESSION)
+    void testRackAwareConnect() throws Exception {
+        installClusterOperator(NAMESPACE, CO_OPERATION_TIMEOUT_SHORT);
+
+        String wrongRackKey = "wrong-key";
+        String rackKey = "rack-key";
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3)
+                .editSpec()
+                    .editKafka()
+                        .withNewRack()
+                            .withTopologyKey(rackKey)
+                        .endRack()
+                        .addToConfig("replica.selector.class", "org.apache.kafka.common.replica.RackAwareReplicaSelector")
+                    .endKafka()
+                .endSpec().done();
+
+        KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+        String kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+        LOGGER.info("Deploy KafkaConnect with wrong rack-aware topology key: {}", wrongRackKey);
+        KafkaConnectResource.kafkaConnectWithoutWait(KafkaConnectResource.defaultKafkaConnect(CLUSTER_NAME, CLUSTER_NAME, 1)
+                .editSpec()
+                    .withNewRack()
+                        .withTopologyKey(wrongRackKey)
+                    .endRack()
+                .endSpec().build());
+
+        PodUtils.waitForPendingPod(CLUSTER_NAME + "-connect");
+        List<String> connectWrongPods = kubeClient().listPodNames(Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND);
+        String connectWrongPodName = connectWrongPods.get(0);
+        LOGGER.info("Waiting for ClusterOperator to get timeout operation of incorrectly set up KafkaConnect");
+        Thread.sleep(CO_OPERATION_TIMEOUT_SHORT + 10000);
+
+        KafkaConnect kc = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get();
+        Condition kcCondition = kc.getStatus().getConditions().get(0);
+        assertThat("NotReady", is(kcCondition.getType()));
+        assertThat("TimeoutException", is(kcCondition.getReason()));
+
+        PodStatus kcWrongStatus = kubeClient().getPod(connectWrongPodName).getStatus();
+        assertThat("Unschedulable", is(kcWrongStatus.getConditions().get(0).getReason()));
+        assertThat("PodScheduled", is(kcWrongStatus.getConditions().get(0).getType()));
+        assertThat(kcWrongStatus.getConditions().get(0).getMessage(), containsString("didn't match node selector"));
+        KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
+        PodUtils.deletePodWithWait(connectWrongPodName);
```

Comment:
Maybe if you already start with a wrong rack, you could also just change the Connect CR and see that it recovers instead of deleting it?

Reply:
I was thinking about it while writing this test. After your suggestion, I think that probably it should be a different test. 
Or even better this test split into two. 
* Positive case (second portion of test) 
* Negative case (incorrect rack-key) with happy-ending (edit pf CR as you suggest)

WDYT @Frawless @scholzj

Reply:
Sounds good to me.

Reply:
Ye, sounds reasonable. What about recovery from the wrong rack? Isn't it something which could be useful for KafkaRoller coverage?

Reply:
What do you mean by recovery from wrong rack? @Frawless - Is this question targeted at me or @scholzj ?
I added both tests now into PR. Locally they are passing, but I think they will fail in this PR's as we will miss rack-key label on nodes...

Reply:
You deploy it with the wrong one => check it fails => fix it to the correct one in the CR => check that the operator recovers from it and fixes it.

Reply:
Yup, that's what I did I believe in `testRackAwareConnectWrongDeployment` :-)